src/main/java/org/apache/xml/security/utils/XMLUtils.java

```diff
@@ -706,6 +706,27 @@ public static Element selectXencNode(Node sibling, String nodeName, int number)
         return null;
     }
 
+    /**
+     *
+     * @param sibling
+     * @param nodeName
+     * @param number
+     * @return nodes with the given node name
+     */
+    public static Element selectXenc11Node(Node sibling, String nodeName, int number) {
+        while (sibling != null) {
+            if (EncryptionConstants.EncryptionSpec11NS.equals(sibling.getNamespaceURI())
+                    && sibling.getLocalName().equals(nodeName)) {
+                if (number == 0){
+                    return (Element)sibling;
```

Comment:
Is it always safe to assume that this is an `Element`? `Text` nodes also implement `getNextSibling()` (for multiple `CDATA` I assume) - but they can never have a namespace URI... Just doesn't feel too perfect. Maybe some comment may help :)

Reply:
For the code readability, I tried to use as much as possible the same code patterns as they have already been used and have been working OK for the last two decades. This approach aims to minimize changes as PR is already large enough so that the focus would be more on the secure implementation of KeyAgreement rather than coding flavors and refactoring. But we can do another PR, if @coheigea  agree with it;  to refactor XMLUtils and other DOM processing methods.